### PR TITLE
Logging fixes

### PR DESCRIPTION
**Pull request by leoetlino**

### CODE REVIEW

File: Source/Core/Common/Logging/LogManager.cpp [hunk 1]
```diff
@@ -100,18 +100,33 @@ LogManager::LogManager()
...
     bool enable;
     logs->Get(container->GetShortName(), &enable, false);
     container->SetEnable(enable);
+    container->SetLevel(static_cast<LogTypes::LOG_LEVELS>(verbosity));
     if (enable && write_file)
       container->AddListener(LogListener::FILE_LISTENER);
     if (enable && write_console)
       container->AddListener(LogListener::CONSOLE_LISTENER);
+    if (enable && write_window)
+      container->AddListener(LogListener::LOG_WINDOW_LISTENER);
```
Comment (shuffle2) [inline Source/Core/Common/Logging/LogManager.cpp]: what does this do on nogui?

Comment (leoetlino) [inline Source/Core/Common/Logging/LogManager.cpp]: ...crash. Whoops, looks like I missed that no window listener would be registered with NoGUI.

I have added a check to make sure listeners exist before using them.

File: Source/Core/Common/Logging/LogManager.cpp [hunk 2]
```diff
@@ -100,18 +100,33 @@ LogManager::LogManager()
...
   options->Get("WriteToFile", &write_file, false);
   options->Get("WriteToConsole", &write_console, true);
+  options->Get("WriteToWindow", &write_window, true);
+
+  // Set up log listeners
+  int verbosity;
+  options->Get("Verbosity", &verbosity, 0);
+
+  // Ensure the verbosity level is valid
+  if (verbosity < 1)
```
Comment (iwubcode) [inline Source/Core/Common/Logging/LogManager.cpp]: I can't recall, do we have access to c++ 17 features?  If so, we could leverage std::clamp

Comment (leoetlino) [inline Source/Core/Common/Logging/LogManager.cpp]: Unfortunately not yet :(
We do have a MathUtils::Clamp though.

Comment (iwubcode) [inline Source/Core/Common/Logging/LogManager.cpp]: That's a shame about c++ 17.  Is MathUtils::Clamp going to be replaced by std::clamp when ready, does it make sense to use here?

File: Source/Core/Common/Logging/LogManager.cpp (hunk 1 above)
Comment (iwubcode) [inline Source/Core/Common/Logging/LogManager.cpp]: Sorry, is that change in this PR?

Comment (leoetlino) [inline Source/Core/Common/Logging/LogManager.cpp]: You mean the check? Yes, it's the last commit (dd8dcdf)

Comment (iwubcode) [inline Source/Core/Common/Logging/LogManager.cpp]: Oh sorry, I was under the impression it was the `container->AddListener(log_window_listener)` that was causing the crash

**Comment by shuffle2** [comment]
This code is split like this because non-WX has no idea what "window" log listener is. I would prefer to keep that separation unless there is a good reason(?). The sanity checking could be moved, tho.

**Comment by spycrab** [comment]
The implementation of the Qt logging widget depends on this.

File: Source/Core/Common/Logging/LogManager.cpp (hunk 2 above)
Comment (lioncash) [inline Source/Core/Common/Logging/LogManager.cpp]: Yeah, it'll eventually be replaced

**Comment by shuffle2** [comment]
Its still gross

**Pull request merged**